src/types/keys.ts

```diff
@@ -0,0 +1,164 @@
+import { Transaction } from "stellar-base";
+
+export enum KeyType {
+  ledger = "ledger",
+  plainTextKey = "plainTextKey",
+}
+
+interface BaseKey {
+  publicKey: string;
+  extra?: string;
```

Comment:
Thinking about this, there's a reason I didn't use inheritance to model a key in the first version.

It comes down to the way a `KeyStore` is implemented. A `KeyStore` will usually need to serialize the key in some way, say by writing it to a database with one column per field. If we lay out a single `Key` interface that defines all the possible fields that we expect a `Key` to need, plus an `extra` field to handle exceptional cases, then a `KeyStore` implementer can lay out an actual database (or other) schema for storing the `Key`. We want a `KeyStore` to be able to handle new kinds of keys it wasn't originally written to handle. If we follow this inheritance approach, the `KeyStore` implementer has two options:

1. JSON encode the whole `Key` and treat it as a blob. This works, but they lose any schema on their end.
2. Switch on the `type` and handle each `type` of key differently. This allows a schema for key types the author is aware of, but new key types that get plugged in later will have unknown fields, so the `KeyStore` would have to store those separately in some blob store.

So explaining all of that, a single `Key` object with an `extra` field seems like the best of both worlds. You get to have a schema, and the `extra` field should be of type `any`. The `KeyStore` implementer can JSON encode `extra` before storing it in its own column, allowing extensibility to new unknown key types.